### PR TITLE
Adds e2e build deploy job with kuttl tests for cinder

### PR DESCRIPTION
**Pull request by frenzyfriday**
- Adding e2e build_deploy_kuttl job (similar to build_deploy but with kuttl)
- Changed always run for the build_deploy job to false

### CODE REVIEW

**Comment by cescgina** [comment]
@frenzyfriday hey, I just checked the last changes. We should modify the `always_run: false` parameter for the `cinder-operator-build-deploy`, not the kuttl one, and both should have `optional: true`. `always_run` controls whether the job will run for every PR, `optional` controls whether the job passing is required for merging (i.e if the job is voting). We want the jobs running for every PR, but not voting, sorry for the confusion

File: ci-operator/jobs/openstack-k8s-operators/cinder-operator/openstack-k8s-operators-cinder-operator-master-presubmits.yaml [hunk 1]
```diff
@@ -62,6 +62,68 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )cinder-operator-build-deploy,?($|\s.*)
```
[the remark this answers is not in the document]
Comment (frenzyfriday) [reply ci-operator/jobs/openstack-k8s-operators/cinder-operator/openstack-k8s-operators-cinder-operator-master-presubmits.yaml:64]: Thanks for clearing up the difference between optional and always run! I thought somehow the optional flag controls whether the test runs on every PR. Updating to always_run: false.

**Review by cescgina — APPROVED** [review]
Thanks for the changes, everything looks good to me

**Comment by rlandy** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-cinder-operator-master-cinder-operator-build-deploy-kuttl

File: ci-operator/jobs/openstack-k8s-operators/cinder-operator/openstack-k8s-operators-cinder-operator-master-presubmits.yaml [hunk 2]
```diff
@@ -62,6 +62,68 @@ presubmits:
...
+    - ^master$
+    - ^master-
+    cluster: build04
+    context: ci/prow/cinder-operator-build-deploy-kuttl
+    decorate: true
+    labels:
+      ci.openshift.io/generator: prowgen
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openstack-k8s-operators-cinder-operator-master-cinder-operator-build-deploy-kuttl
+    optional: true
```
Comment (marios) [inline ci-operator/jobs/openstack-k8s-operators/cinder-operator/openstack-k8s-operators-cinder-operator-master-presubmits.yaml:77]: don't we want to block on kuttl test failures though?

**Review by marios — CHANGES_REQUESTED** [review]
sanity check on optional otherwise lgtm

File: ci-operator/jobs/openstack-k8s-operators/cinder-operator/openstack-k8s-operators-cinder-operator-master-presubmits.yaml (hunk 2 above)
Comment (frenzyfriday) [reply ci-operator/jobs/openstack-k8s-operators/cinder-operator/openstack-k8s-operators-cinder-operator-master-presubmits.yaml:77]: I think we are keeping all the kuttl workflows non voting for now, maybe till they are stable or till the demo?
Example:
- https://github.com/openshift/release/pull/37275/files#diff-285068ae4683b9b46983eb9940594dd82e41f5d6e83ba2bf7331a1616d9c145fR97
- https://github.com/openshift/release/pull/37103/files#diff-f519f0f7af9c285191571fe9eca8d6a45324fa0a235bb8538b1bd593890ec147R97

**Review by rlandy — COMMENTED** [review]
lgtm - pls make tests voting - no optional in next review

**Pull request merged**